### PR TITLE
refactor: convert block API to async await

### PR DESCRIPTION
**Pull request by alanshaw**
depends on:

* [x] #1150

### CODE REVIEW

File: src/block/put.js [hunk 1]
```diff
@@ -1,74 +1,60 @@
...
-    if (block.cid || options.cid) {
-      let cid
-
-      try {
-        cid = new CID(block.cid || options.cid)
-      } catch (err) {
-        return callback(err)
+    const searchParams = new URLSearchParams(options.searchParams)
+    if (options.format) searchParams.set('format', options.format)
+    if (options.format) searchParams.set('mhtype', options.mhtype)
```
Comment (hugomrdias) [inline src/block/put.js]:
```suggestion
    if (options.mhtype) searchParams.set('mhtype', options.mhtype)
```

Comment (alanshaw) [reply src/block/put.js]: Good catch, thanks!

**Comment by alanshaw** [comment]
```js
const searchParams = new URLSearchParams()
```

Agree, that shouldn't happen, we should allow them to be overridden with options on every API call. Along with `timeout`, `signal` etc.

```js
const searchParams = new URLSearchParams(options.searchParams)
```

`URLSearchParams` allows you to pass a `URLSearchParams` instance OR an object as the parameter. This allows _users_ to pass their own query string params as an object or an instance of `URLSearchParams` AND also allows _us_ to add mandatory/overridden parameters from args or other options.

```js
if (options.format !== undefined) {
  searchParams.set('format', options.format)
}

if(options.format != null)
```

Sometimes we check for `null`/`undefined` because a falsy check on booleans or numbers that might be 0 is not good enough. It's a matter of preference if we only consider `undefined` as a "value" that doesn't get added to the query string. I've been using the `!= null` because it includes `null` as well as `undefined` and 99.9% of the time you also would not want `?arg=null` which is what happens when you pass `null` to `URLSearchParams` and stringify it.

**Comment by alanshaw** [comment]
@hugomrdias apologies I'm going to expand this PR to include the DAG API since it uses the block API.

**Comment by achingbrain** [comment]
I think we should apply the robustness principal - that is, all the `if` statements around `URLSearchParams` are verbose but it's good to only send what we know the server will support.

**Pull request merged**